FileSpecs/Events.md

```diff
@@ -0,0 +1,265 @@
+|               |       | `12` | U12.1 | U12.2 | U12.3 | U12.4 | `00`   | `00`  | U12.7   | `00`  | U12.9 |
+|               |       | `13` | U13.1 | U13.2 | `00`  | U13.4 | U13.5  | U13.6 | U13.7   | `00`  | U13.9 |
+| **Door3D**    | 3D    | `14` | `01`  | `00`  | `00`  | U14.4 | U14.5  | `00`  | U14.7   | `00`  | `00`  |
+|               | 2D    | `15` | U15.1 | U15.2 | U15.3 | `00`  | `00`   | `00`  | U15.7   | `00`  | `00`  |
+|               | 2D    | `16` | `00`  | `00`  | `00`  | `00`  | `00`   | `00`  | `00`    | `00`  | `00`  |
+| **Win**       | 2D    | `17` | `00`  | `00`  | `00`  | `00`  | `00`   | `00`  | `00`    | `00`  | `00`  |
+
+
+* Column `Where` indicates whether the event only appears in 2D or 3D maps.
+* Parameters of the form "U00.0" indicate an unknown parameter.
+* `00` indicates that this paramter always appears as `00`.
+
+This information is observational: the Amberstar *engine* might support more options values, but the *game* does not seeem to use them (unless events can be triggered in other ways than observed so far).
+
+<!-- ---------------------------------------- -->
+## Event 01: Teleport
+
+Teleports the party to the specified *map* - 1, coordinates (*x*, *y*).
+
+Unknown: *map*=0 might teleport the player to a different location in the same map (?)
+
+### Unknowns
+* U01.3: [`00-03`]
+
+<!-- ---------------------------------------- -->
+## Event 02: Door
+
+Party encounters a locked door.  The lock-picking difficulty seems to be indicated by *cr*, which is 1, 10, or 100.
+
+### Unknowns
+* Exact meaning of the challenge rating *cr*
+* U02.2: [`00`, `01`, `03` (only 3D maps)]
+* U02.3: [`00`, `0a`], on 3D maps also possibly [`05`, `0f`, `23`]
+* U02.5: [`01-0f`], but not `0c-0d`.
+* U02.6: `00`, but may be `01` on 3D maps
+* U02.7: `00` or special number, likely the key ID?  Is nonzero iff *cr* = 100
+
+<!-- ---------------------------------------- -->
+## Event 03: Message
+*Trigger*: Entering, if U03.3 is 0, or Looking if U03.3 is 1.  (Other values for U03.3: Unknown.)
+
+Displays message *msg*, from the local `MAPTEXT.AMB` resources, and possibly shows image *img* - 1 from `PICS80.AMB`, if *img* is nonzero.
+The party learns the dictionary keyword *kw* for use in conversation with NPCs.
+
+### Unknowns
+* U03.3: [`00`, `02`, `07`, `09`, `0a`, `14`, `19`, `32`] (migt be flags?)
+* U03.5: [`00-28`]
+
+<!-- ---------------------------------------- -->
+## Event 04: Chest
+*Trigger*: Looking
+
+The party accesses the that contains the items from `CHESTDAT.AMB` resource number *chest*, while displaying message *msg*.  Once the chest is empty,
+the game records this fact with chest flag *flag* and no longer allows chest access.
+
+The gold for this chest is stored in u16 values in `PARTYDAT.SAV`, starting at `00001ef0` for chest `01`.
+
+### Unknowns
+* U04.1: [`00`, `01`, `0a`]: Lockpick difficulty?  `0a` only in 3D maps.
+* U04.2: [`00-07`]
+* U04.3: [`00`], or [`01`, `05`] in 3D maps
+* U04.4: [`00-2a`]
```

Comment:
One of those unknown bytes might be a text index which is shown when opening the chest? In Ambermoon there is also the possibility to specify an event which is triggered when failing lockpicking. And also a key index should be possible I guess.

Reply:
The message goes in the *msg* parameter.  I haven't tested chests with lockpicking yet-- so far I haven't seen any event-chaining control flow as in Ambermoon, but most of the event descriptions are unknowns, so there is plenty of space for that. :-)